src/gui/editors/AutomationEditor.cpp

```diff
@@ -1314,8 +1314,9 @@ void AutomationEditor::paintEvent(QPaintEvent * pe )
 		m_leftRightScroll->setPageStep( l );
 	}
 
-	if( validPattern() )
+	if( validPattern() && GuiApplication::instance()->automationEditor()->hasFocus() )
```

Comment:
The name is confusing. ````!GuiApplication::instance()->automationEditor()```` returns an ````AutomationEditorWindow````.

It's this ````AutomationEditorWindow````.that we need to check not ````AutomationEditor````. It took me a while to figure this out.